docs: fix grammar in crate structure (Qiskit's/Rust/run the tests)

ManuelEsparcia opened the pull request:
Fix 3 issues in the “Crate structure” section:

- more of Qiskit functionality → more of Qiskit's functionality
- the rust code → the Rust code
- run with the tests with that disabled → run the tests with that disabled

Docs-only change; no behavior impact.
Signed-off-by: Manuel Esparcia <manuelesparciacantos@gmail.com>

ManuelEsparcia committed:
Fix typos and enhance README clarity

Fix 3 issues:
- more of Qiskit functionality → more of Qiskit's functionality
- the rust code → the Rust code
- run with the tests with that disabled → run the tests with that disabled

Docs-only change; no behavior impact.
Signed-off-by: Manuel Esparcia <manuelesparciacantos@gmail.com>

jakelishman commented on crates/README.md:
````diff
@@ -59,15 +59,7 @@ We can revisit that if we start writing crates that we intend to publish.
 
 ```bash
 cargo test --no-default-features
-```
-
-On Linux, you might find that the dynamic linker still can't find `libpython`.
-If so, you can extend the environment variable `LD_LIBRARY_PATH` to include it:
-
-```bash
-export LD_LIBRARY_PATH="$(python -c 'import sys; print(sys.base_prefix)')/lib:$LD_LIBRARY_PATH"
-```
````
Was this accidental?

jakelishman commented on crates/README.md:
````diff
@@ -59,15 +59,7 @@ We can revisit that if we start writing crates that we intend to publish.
+By default, we activate PyO3's `extension-module` feature in `crates/pyext`, so you have to run the tests with that disabled:
 
 ```bash
 cargo test --no-default-features
````
tbh this whole section is out of date - it's easier just to do
```bash
python tools/run_cargo_test.py
```
which should take care of everything (including the deleted bit below)

ManuelEsparcia committed:
docs: modernize “Running the tests” and small grammar fixes

Follow maintainer recommendation to use `python tools/run_cargo_test.py`
and remove the outdated LD_LIBRARY_PATH note. Keep a minimal
`cargo test --no-default-features` alternative. Also fix “RUST” → “Rust”.

Signed-off-by: Manuel Esparcia <tu-email@ejemplo.com>

ManuelEsparcia committed:
Merge branch 'main' into docs/crate-structure-typos

ManuelEsparcia committed:
Merge branch 'main' into docs/crate-structure-typos

ManuelEsparcia committed:
Merge branch 'main' into docs/crate-structure-typos
